bundler/lib/bundler/current_ruby.rb

```diff
@@ -71,9 +83,21 @@ def windows?
         RUBY_VERSION.start_with?("#{version}.")
       end
 
-      all_platforms = PLATFORM_MAP.keys << "maglev"
```

Comment:
I would've tried a slightly different implementation of not adding "maglev" here and instead define the methods separately:

```ruby
define_method(:"maglev_#{trimmed_version}?") do
  message =
    "`CurrentRuby##{__method__}` is deprecated with no replacement. Please use the " \
    "built-in Ruby `RUBY_ENGINE` and `RUBY_VERSION` constants to perform a similar check."
  removed_message =
    "`CurrentRuby##{__method__}` was removed with no replacement. Please use the " \
    "built-in Ruby `RUBY_ENGINE` and `RUBY_VERSION` constants to perform a similar check."

  SharedHelpers.major_deprecation(2, message, removed_message: removed_message, print_caller_location: true)

  maglev? && send(:"on_#{trimmed_version}?")
end
```

I think it's a bit simpler to parse.

Reply:
👍 I'll change